app/assets/src/components/views/amr_heatmap/AMRHeatmapView.jsx

```diff
@@ -116,6 +130,14 @@ export default class AMRHeatmapView extends React.Component {
       });
     });
 
+    // Append a number to a sample's name to differentiate between samples with the same name.
+    duplicateSampleNames.forEach(sampleName => {
+      for (let i = 0; i < sampleNames.get(sampleName).length; i++) {
+        let sample = sampleNames.get(sampleName)[i];
```

Comment:
I just realized that you need to get the sample from `filteredSamples` in order to modify the name used by the AMR Heatmap. That's the only thing returned by the function, `sampleNames` gets discarded. You only need to modify the sample names that are duplicated, so there's no need to track every single sample name.

Reply:
I'm not sure what you're suggesting. Wouldn't I need to iterate through `filteredSamples` to find the duplicated sample names, then do a second pass to update them?

I figured that keeping a set of `duplicatedSampleNames` and a map of {sampleName: [samples with sampleName]}, then just iterating through the `duplicatedSampleNames` would be the quickest way to update the names to prevent a second pass through all of `filteredSamples`. Sorry if I misunderstood your suggestion! 😅

Reply:
You could just keep track of the number of duplicates and immediately set the the suffix given the current count.
You would only need to do one cycle.

Reply:
I would need to do another pass afterwards to add (1) to the first sample found with that name, since I wouldn't know it was a duplicate during the first cycle.

Reply:
AFAICT what happens in the `sampleNames` object isn’t returned or otherwise used in the component’s state, so once the function is done executing and the object goes out of scope it’s deleted.

Reply:
If you look at the `requestSampleData` function you can see the processing pipeline and where `filteredSamples` is used.

Reply:
You’re right that you have to track every name for dupes, my bad

Reply:
Ah, they all get a suffix ?

Reply:
Yeah, that's the pattern I went with for the taxon heatmap as well. Looping in @happyimadesignr to double check!

Reply:
Ohh you’re storing a reference to the sample in the Map. I see that now